pjsip/src/pjsip/sip_transaction.c

```diff
@@ -830,6 +862,50 @@ static pj_status_t mod_tsx_layer_unload(void)
 }
 
 
+/* Detect merged requests as per RFC 3261 section 8.2.2.2. */
+PJ_DEF(pjsip_transaction *)
+pjsip_tsx_detect_merged_requests(pjsip_rx_data *rdata)
+{
+    pj_str_t key, key2;
+    pj_uint32_t hval = 0;
+    pj_status_t status;
+
+    /* If the request has no tag in the To header field, the UAS core MUST
+     * check the request against ongoing transactions.
+     */
+    if (rdata->msg_info.to->tag.slen != 0)
+        return NULL;
+
+    status = pjsip_tsx_create_key(rdata->tp_info.pool, &key, PJSIP_ROLE_UAS,
+                                  &rdata->msg_info.cseq->method, rdata);
+    if (status != PJ_SUCCESS)
+        return NULL;
```

Comment:
write log for debugging?

Reply:
why though? it's very rare for create key to fail, the only possibility being out of memory. in fact, some places don't even check the return status, even in a critical location such as `mod_tsx_layer_on_rx_request`:
https://github.com/pjsip/pjproject/blob/master/pjsip/src/pjsip/sip_transaction.c#L842

same below.

Reply:
Nothing urgent/major, just come to my mind while implementing this, adding one log line may help debugging as it looks like we cannot differentiate when the function returning `NULL`, is it because the merged req tsx is not found or something fails in the processing (create key, etc).